test/CadlRanchProjects.Tests/azure-core-basic.cs

```diff
@@ -99,5 +99,35 @@ await foreach (Page<User> page in allPages.AsPages())
             User response = await new BasicClient(host, null).ExportAsync(1, "json");
             Assert.AreEqual("Madge", response.Name);
         });
+
+        [Test]
+        public Task Azure_Core_Basic_listWithParameters() => Test(async (host) =>
+        {
+            AsyncPageable<User> allPages = new BasicClient(host, null).GetWithParametersAsync(new ListItemInputBody("Madge"), ListItemInputExtensibleEnum.Second);
+            await foreach (Page<User> page in allPages.AsPages())
```

Comment:
we should just foreach over all the result, such as
```suggestion
            await foreach (var item in allPages)
```